packages/netlify-cms-widget-markdown/src/serializers/remarkShortcodes.js

```diff
@@ -10,6 +10,33 @@ export function remarkParseShortcodes({ plugins }) {
 
 function createShortcodeTokenizer({ plugins }) {
   return function tokenizeShortcode(eat, value, silent) {
+    // Plugin patterns may rely on `^` and `$` tokens, even if they don't
+    // use the multiline flag. To support this, we fall back to searching
+    // through each line individually, trimming trailing whitespace and
+    // newlines, if we don't initially match on a pattern. We keep track of
+    // the starting position of each line so that we can sort correctly
+    // across the full multilen matches.
+    const trimmedLines = value
+      .split('\n\n')
+      .reduce((acc, line) => {
+        const [
+          { start: previousLineStart, originalLength: previousLineOriginalLength } = {
+            start: 0,
+            originalLength: 0,
+          },
+        ] = acc;
+        return [
+          {
+            line: line.trimEnd(),
+            start: previousLineStart + previousLineOriginalLength + 2,
+            originalLength: line.length,
+          },
+          ...acc,
+        ];
+      }, [])
+      .reverse()
+      .map(({ line, start }) => ({ line, start }));
+
```

Comment:
@erezrokah I don't know how you all like to handle breaking changes, but I think dropping support for this behavior would be a great candidate for the next major version release! Especially now that it's not mentioned in the documentation. We would also have to update the pattern for the built-in image shortcode to turn on the multiline flag.